cli/tests/test_diff_command.rs

```diff
@@ -2843,3 +2845,90 @@ fn test_diff_binary() {
     [EOF]
     ");
 }
+
+#[test]
+fn test_diff_revisions() {
+    let test_env = TestEnvironment::default();
+    test_env.run_jj_in(".", ["git", "init", "repo"]).success();
+    let repo_path = test_env.env_root().join("repo");
+
+    //
+    //
+    //
+    // E
+    // |\
+    // C D
+    // |/
+    // B
+    // |
+    // A
+    create_commit(&test_env, &repo_path, "A", &[]);
+    create_commit(&test_env, &repo_path, "B", &["A"]);
+    create_commit(&test_env, &repo_path, "C", &["B"]);
+    create_commit(&test_env, &repo_path, "D", &["B"]);
+    create_commit(&test_env, &repo_path, "E", &["C", "D"]);
+
+    let diff_revisions = |expression: &str| -> CommandOutput {
+        test_env.run_jj_in(&repo_path, ["diff", "-s", "-r", expression])
+    };
+    // Can diff a single revision
+    insta::assert_snapshot!(diff_revisions("B"), @r"
+    A B
```

Comment:
nit: use `--name-only`? `A` here is confusing.

Reply:
Done.